heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraHeronShellController.java

```diff
@@ -74,26 +74,38 @@ public boolean killJob() {
   public boolean restart(Integer containerId) {
     // there is no backpressure for container 0, delegate to aurora client
     if (containerId == null || containerId == 0) {
-      cliController.restart(containerId);
+      return cliController.restart(containerId);
     }
 
     if (stateMgrAdaptor == null) {
       LOG.warning("SchedulerStateManagerAdaptor not initialized");
       return false;
     }
 
-    int index = containerId - 1; // stmgr container starts from 1
-    StMgr contaienrInfo = stateMgrAdaptor.getPhysicalPlan(topologyName).getStmgrs(index);
-    String host = contaienrInfo.getHostName();
-    int port = contaienrInfo.getShellPort();
+    String host = "x";
+    int port = -1;
+    int offset = "stmgr-".length();
+    for (StMgr sm : stateMgrAdaptor.getPhysicalPlan(topologyName).getStmgrsList()) {
+      Integer id = Integer.valueOf(sm.getId().substring(offset));
+      LOG.info("comparing container " + offset + " " + sm.getId() + " with " + containerId);
+      if (containerId.equals(id)) {
+        host = sm.getHostName();
+        port = sm.getShellPort();
+        break;
```

Comment:
What if the id is not found? It would be better to have protective code instead of creating a connection to "http://x:-1/killexecutor".

Reply:
done